api/room/schema.py

```diff
@@ -88,11 +88,18 @@ def mutate(self, info, office_id, **kwargs):
 
         admin_roles.create_rooms_update_delete_office(office_id)
         query = Room.get_query(info)
-        exact_query = room_join_office(query)
+        active_rooms = query.filter(RoomModel.state == "active")
```

Comment:
Is there a way you can combine the query for both state and calendar id in one query statement and obtain a particular room object directly with these attributes? And since we will be getting only one room object from our query result, do we need to check the count?

Reply:
I have made improvements to this query @Bonifase  After querying for active rooms, I loop through the returned active rooms to check which active room has a calendar id that has been provided.